src/ElasticsearchRequestWrapper.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace Drupal\elasticsearch_helper;
+
+use Drupal\elasticsearch_helper\Event\ElasticsearchEvents;
+use Drupal\elasticsearch_helper\Event\ElasticsearchOperationRequestEvent;
+use Drupal\elasticsearch_helper\Event\ElasticsearchOperationRequestResultEvent;
+use Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexInterface;
+
+/**
+ * Class ElasticsearchRequestWrapper
+ */
+class ElasticsearchRequestWrapper {
+
+  /**
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+  /**
+   * Elasticsearch operation.
+   *
+   * @var string
+   */
+  protected $operation;
+
+  /**
+   * Elasticsearch operation request callable.
+   *
+   * @var callable
+   */
+  protected $callback;
+
+  /**
+   * Elasticsearch operation request callable parameters.
+   *
+   * @var array
+   */
+  protected $callbackParameters = [];
+
+  /**
+   * Elasticsearch index plugin instance.
+   *
+   * @var \Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexInterface
+   */
+  protected $pluginInstance;
+
+  /**
+   * Index-able object.
+   *
+   * @var mixed|null
+   */
+  protected $object;
+
+  /**
+   * ElasticsearchRequestWrapper constructor.
+   *
+   * @param $operation
+   * @param $callback
+   * @param array $callback_parameters
+   * @param \Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexInterface $plugin_instance
+   * @param null $object
```

Comment:
Can you add description for these type declarations? It might be helpful later :)